zanata-war/src/main/java/org/zanata/rest/service/GlossaryService.java

```diff
@@ -88,37 +207,90 @@ public Response get(LocaleId locale) {
 
     @Override
     public Response put(Glossary glossary) {
-        identity.checkPermission("", "glossary-insert");
-        ResponseBuilder response;
+        return insertOrUpdate(glossary);
+    }
+
+    @Override
+    public Response post(Glossary glossary) {
+        return insertOrUpdate(glossary);
+    }
```

Comment:
@aeng Based on the conversation we had, here's what I would recommend:

1. Lets leave the current endpoints that deal with entire Glossaries to do exactly that. They either create or update an **entire** glossary, not just the given entries. These would continue to reside under the `/glossary` url.

2. Lets create other endpoints for individual glossary entries: The would deal with the individual entries and would be able to filter by language, sort, etc. The `POST` and `PUT` would also be able to deal with updating or creating individual entries. Pretty much the additional logic you have added.

The problem that I have with the current setup is that it's not really RESTful. Our `POST` calls are just modifying some of the entries and the whole resource that's being passed. This sounds like a prime example of a sub-resource instead.

we can talk about this a bit more face-2-face. Might want to include @seanf in this discussion as well.